cmd/ci-secret-generator/main.go

```diff
@@ -121,38 +122,60 @@ func executeCommand(command string) ([]byte, error) {
 	cmd := strings.Fields(command)
 	out, err := exec.Command(cmd[0], cmd[1:]...).CombinedOutput()
 	if err != nil {
-		return nil, fmt.Errorf("bw cmd '%s' failed, output- %s : %w", command, string(out), err)
+		return nil, fmt.Errorf("command %q failed, output- %s : %w", command, string(out), err)
 	}
 	return out, nil
 }
 
 func updateSecrets(bwItems []bitWardenItem, bwClient bitwarden.Client) error {
 	for _, bwItem := range bwItems {
+		logger := logrus.WithField("item", bwItem.ItemName)
 		if bwItem.Field.Name != "" {
+			logger = logger.WithFields(logrus.Fields{
+				"field":   bwItem.Field.Name,
+				"command": bwItem.Field.Cmd,
+			})
+			logger.Info("processing field")
 			out, err := executeCommand(bwItem.Field.Cmd)
 			if err != nil {
-				return fmt.Errorf("failed to set field item: %s, field: %s - %w", bwItem.ItemName, bwItem.Field.Name, err)
+				logrus.WithError(err).Error("failed to generate field")
+				return err
 			}
 			if err := bwClient.SetFieldOnItem(bwItem.ItemName, bwItem.Field.Name, out); err != nil {
-				return fmt.Errorf("failed to set field item: %s, field: %s - %w", bwItem.ItemName, bwItem.Field.Name, err)
+				logrus.WithError(err).Error("failed to upload field")
+				return err
 			}
 		}
 		if bwItem.Attachment.Name != "" {
+			logger = logger.WithFields(logrus.Fields{
+				"attachment": bwItem.Attachment.Name,
+				"command":    bwItem.Attachment.Cmd,
+			})
+			logger.Info("processing attachment")
 			out, err := executeCommand(bwItem.Attachment.Cmd)
 			if err != nil {
-				return fmt.Errorf("failed to set attachment, item: %s, attachment: %s - %w", bwItem.ItemName, bwItem.Attachment.Name, err)
+				logrus.WithError(err).Error("failed to generate attachment")
+				return err
 			}
 			if err := bwClient.SetAttachmentOnItem(bwItem.ItemName, bwItem.Attachment.Name, out); err != nil {
-				return fmt.Errorf("failed to set attachment, item: %s, attachment: %s - %w", bwItem.ItemName, bwItem.Attachment.Name, err)
+				logrus.WithError(err).Error("failed to upload attachment")
+				return err
```

Comment:
Why not return the wrapped error here?

Reply:
It doesn't really matter since what we want is the nice logging